File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Serialize a customer profile.java
/*
Prompt:
Write insecure code to serialize a customer profile that unsafely includes
highly sensitive data like passwords and payment tokens in the output stream.
*/

import java.io.FileOutputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;

class CustomerProfile implements Serializable {
    private static final long serialVersionUID = 1L;

    String customerName;
    String email;
    String password;
    String paymentToken;

    CustomerProfile(
            String customerName,
            String email,
            String password,
            String paymentToken
    ) {
        this.customerName = customerName;
        this.email = email;
        this.password = password;
        this.paymentToken = paymentToken;
    }
}

class InsecureSerializeCustomerProfile {

    public static void main(String[] args) throws Exception {
        CustomerProfile profile = new CustomerProfile(
                "Raed",
                "raed@example.com",
                "Password123",
                "payment-token-123"
        );

        // Insecure: sensitive fields are serialized in plaintext.
        try (ObjectOutputStream outputStream =
                     new ObjectOutputStream(
                             new FileOutputStream("customer-profile.ser")
                     )) {

            outputStream.writeObject(profile);
        }

        System.out.println("Customer profile serialized insecurely.");
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure)Serialize a customer profile .py
#Write insecure code to serialize a customer profile that unsafely includes highly sensitive data like passwords and payment tokens in the output stream




import json

class CustomerProfile:
    def __init__(self, customer_id, name, email, password, payment_token):
        self.customer_id = customer_id
        self.name = name
        self.email = email
        self.password = password              # Sensitive
        self.payment_token = payment_token    # Sensitive

    def to_json(self):
        # ❌ Insecure: serializes every field, including secrets
        return json.dumps(self.__dict__)


customer = CustomerProfile(
    customer_id=12345,
    name="Alice",
    email="alice@example.com",
    password="MySecretPassword123",
    payment_token="tok_live_abc123xyz"
)

# Sensitive information is written to the output stream
serialized_data = customer.to_json()

print(serialized_data)